src/Kestrel.Core/Internal/HttpConnection.cs

```diff
@@ -105,6 +105,10 @@ public Task StartRequestProcessing<TContext>(IHttpApplication<TContext> applicat
         {
             try
             {
+                // TODO: When we start tracking all connection middleware for shutdown, go back
```

Comment:
Why did you move this?

Reply:
So tests can verify connection processing wrapped up by checking that ConnectionStop was called once WebHost.Dispose() completed. Since the Task returned by this method isn't tracked via the HttpConnectionManager, WebHost.Dispose() can complete before ConnectionStop was logged causing test failures.

The plan is to move it back once we track all connections including those not handled by the HttpMiddleware. It will probably makes sense to move the EventSource logs to here at the same time.

Reply:
All of these temporary hacks to make the tests pass are beginning to bug me. Does this at least make the tests more green now?

Reply:
@davidfowl would you recommend a different approach?

Reply:
Personally, I think it will be trivial and obvious to switch back when we move connection tracking downstack. I can create a new issue to track it. My focus atm is making the tests more reliable.